.github/workflows/test-bug-run-badge.yml

```diff
@@ -0,0 +1,78 @@
+name: Test new bug report badge
+run-name: Test bug report on issue ${{ github.event.issue.number }}
+on:
+  issues:
+    types: [opened]
+jobs:
+  extract-bug-badge-url:
+    if: ${{ contains(github.event.issue.labels.*.name, 'question') }}
+    runs-on: ubuntu-latest
+    outputs:
+      runBadgeTest: ${{ steps.testCondition.outputs.runNext }}
+      link: ${{ steps.testCondition.outputs.link }}
+    steps:
+      - name: Test badge test run conditions
+        id: testCondition
+        run: |
+          product=$(echo "${{ github.event.issue.body }}" | grep -A2 "Are you experiencing an issue with.*" | tail -n 1)
+          link=$(echo "${{ github.event.issue.body }}" | grep -A2 "Link to the badge.*" | tail -n 1)
+
+          if [[ "$product" == "shields.io" && "$link" == "https://img.shields.io"* ]]; then
+            echo "runNext=true" >> "$GITHUB_OUTPUT"
+            echo "link=$link" >> "$GITHUB_OUTPUT"
+          else
+            echo "Conditions not met. Skipping the workflow..."
+            echo "runNext=false" >> "$GITHUB_OUTPUT"
+          fi
+
+  run-bug-badge-url-test:
+    needs: extract-bug-badge-url
+    if: needs.extract-bug-badge-url.outputs.runBadgeTest == 'true'
+    permissions:
+      issues: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup
+        uses: ./.github/actions/setup
+        with:
+          node-version: 16
+          cypress: false
+
+      - name: Run tests
+        id: run_tests
+        env:
```

Comment:
We shouldn't pass this job any secrets.

There is nothing in `npm run badge` to scrub or obfuscate credentials so any credentials we pass in are going to be echoed out in the request details. GH actions does have some built in masking functionality, but it won't cover basic auth credentials for example. They get base64 encoded. Obviously this means there are 9 service (out of ~200) where this will fail but I think we can live with that.

If we want to improve that, I would want to implement something at the logging layer that optionally prevents credentials from being included in the log output at all. Although we would want it to be an optional flag - outputting the credentials is really useful locally. I used it literally yesterday when I was working on https://github.com/badges/shields/pull/9387 to make sure the auth was being passed properly.

Reply:
I will remove the secrets (will commit soon), but i am not sure that only 9 services will be affected.
For example, the github token is there to avoid rate limit with a large number of services.

Regarding avoiding secrets from leaking, seems a bit risky. I think we could open a new issue to get ideas about how to tackle that.

Reply:
GitHub is a "service"
GitHub Issues or GitHub License is a "badge"
So it is one "service", but your core point is completely right: GitHub is a service that has a lot of badges and they are really widely used so it would be very nice if that one worked.

Fortunately, if we're only calling one badge URL we don't really _need_ a token. We need one under test because we're making lots of API calls. Anonymous usage would be fine here because we're not going to hit the late limit by making only one request. Allowing GH badges to work with no GitHub credentials at all would be a great solution. This has actually been a really long-standing issue affecting local development too: https://github.com/badges/shields/issues/2754 I don't think we need to fix that for this workflow to be useful, but if you were looking for a next issue to help with, https://github.com/badges/shields/issues/2754 would be a great shout.

We can merge this as it stands and treat that as a follow up.